gold-cc-cvc-input.html

```diff
@@ -183,10 +184,27 @@
           this._handleAutoValidate();
         },
 
-        _computeRequiredLength: function(cardType) {
+        _onCardTypeChanged: function(cardType, oldCardType) {
+          // The initial property assignment is handled by `ready`.
+          if (oldCardType == undefined)
+            return;
+          if (this.value) this._handleAutoValidate();
+        },
+
+        _computeMaxLength: function(cardType) {
+          if (!cardType) return 4;
           return this._computeIsAmex(cardType) ? 4 : 3;
         },
 
+        _validateLength: function(value, cardType) {
+          var length = value.length || 0;
+          if (!cardType) {
+            return length === 3 || length === 4;
+          } else {
+            return length === this._computeMaxLength(cardType);
```

Comment:
i think `return length === this._maxLength` should work, since it's a computed property